src/components/LazyLoader/helpers.js

```diff
@@ -0,0 +1,71 @@
+import { PLACEHOLDER_IMAGE, DEFAULT_TARGET_DENSITY } from "./constants";
+import createParams from "../../utils/createParams";
+
+/* istanbul ignore next */
+const Url = process.browser ? global.window.URL : require("url");
+
+export const resize = ({ src = "", ...params }) => {
+  try {
+    const { host, pathname } = process.browser ? new Url(src) : Url.parse(src);
+
+    if (!host) {
+      return src;
+    }
+
+    const url = `https://${host}/${pathname}`;
+    const fit = params.width && params.height ? { fit: "crop" } : {};
+    const resizeSrc = `${url}${createParams({ ...params, ...fit })}`;
+    return resizeSrc;
+  } catch (e) {
+    return src;
+  }
+};
+
+export const createGetSrcByDensity = ({
+  src,
+  width,
+  height,
+  resizeFn
+}) => density =>
+  resizeFn({
+    src,
+    width: width ? width * density : null,
+    height: height ? height * density : null
+  });
+
+export const getSrcSetByDensity = (ref, getSrcByDensity) =>
+  `${getSrcByDensity(1)} 1x, ${getSrcByDensity(2)} 2x, ${getSrcByDensity(
+    3
+  )} 3x`;
+
+export const getTargetDensity = (
+  defaultTargetDensity = DEFAULT_TARGET_DENSITY
+) => {
+  if (typeof global.window !== "object" || !global.window.devicePixelRatio) {
+    return defaultTargetDensity;
+  }
+
+  return global.window.devicePixelRatio > 3
```

Comment:
What's the importance of `3` here?

Reply:
I thought it might be the max value for this property, but now I after more investigation it seems that values go up to 5: https://gist.github.com/cheeaun/8838438

I'll adjust the provider logic, so we can resize images to appropriate resolutions for users with higher values.